### PR TITLE
Prepare for 1.11.0 release

### PR DESCRIPTION
**Pull request by ebezzi**
1. Update the `tiledb` dependency for API to ~1.8.0
2. Update R's `DESCRIPTION` file

R docs do not need to be regenerated as they're now part of the CI/CD.

### CODE REVIEW

File: api/python/cellxgene_census/pyproject.toml [hunk 1]
```diff
@@ -31,7 +31,7 @@ dependencies= [
     # NOTE: the tiledbsoma version must be >= to the version used in the Census builder, to
     # ensure that the assets are readable (tiledbsoma supports backward compatible reading).
     # Make sure this version does not fall behind the builder's tiledbsoma version.
-    "tiledbsoma~=1.7.0",
+    "tiledbsoma~=1.8.0",
```
Comment (bkmartinjr) [inline api/python/cellxgene_census/pyproject.toml:34]: I believe tiledbsoma==1.8.0 supports python 3.12.  Do you want to support 3.12 in this release as well, or does that require additional effort?

**Review by bkmartinjr — APPROVED** [review]
One question regarding python 3.12 support, but otherwise LGTM

File: api/python/cellxgene_census/pyproject.toml (hunk 1 above)
Comment (ebezzi) [reply api/python/cellxgene_census/pyproject.toml:34]: Looks like there are some issues with Python 3.12 in tiledbsoma so the support is marked as provisional: https://github.com/single-cell-data/TileDB-SOMA/issues/1849.

So I think it's safe to leave it out from the Census for now. I'll ask John to let us know when it's fully supported.

Comment (bkmartinjr) [reply api/python/cellxgene_census/pyproject.toml:34]: sounds good. Also cross-referencing #839

**Pull request merged**